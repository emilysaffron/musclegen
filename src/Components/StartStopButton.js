import styled from "@emotion/styled";
import { Link } from "@reach/router";

const ControlButton = styled.button`
  width: 50px;
  height: 25px;
  border-radius: 50%;
  &:hover {
    cursor: pointer;
  }
`;

const StartStopButton = ({ control, label, onClick }) => {
  const labelWithoutWhitespace = label.replace(/\s+/g, "-");
  if (control === "start") {
    return (
      <Link to={`workout`}>
        <ControlButton onClick={onClick}> Start</ControlButton>
      </Link>
    );
  }

  if (control === "stop") {
    return (
      <Link to={`${labelWithoutWhitespace.toLowerCase()}`}>
        <ControlButton onClick={onClick}>Stop</ControlButton>
      </Link>
    );
  }
};

export default StartStopButton;
