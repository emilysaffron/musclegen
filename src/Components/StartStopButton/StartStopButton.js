import styled from "@emotion/styled";
import { Link } from "@reach/router";

const ControlButton = styled.button`
  width: 50px;
  height: 25px;
  border-radius: 50%;
  cursor: pointer;
`;

const StartStopButton = ({ control, urlRoute, onClick }) => {
  const url = urlRoute.replace(/\s+/g, "-");
  if (control === "start") {
    return (
      <Link to={`workout`}>
        <ControlButton onClick={onClick}> Start</ControlButton>
      </Link>
    );
  }

  if (control === "stop") {
    return (
      <Link to={`${url.toLowerCase()}`}>
        <ControlButton onClick={onClick}>Stop</ControlButton>
      </Link>
    );
  }
};

export default StartStopButton;
