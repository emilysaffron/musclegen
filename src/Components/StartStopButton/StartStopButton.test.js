import renderer from "react-test-renderer";
import StartStopButton from "./StartStopButton";

it("StartStopButton Snapshot", () => {
  const tree = renderer
    .create(
      <>
        <StartStopButton
          control="start"
          urlRoute="bodyTarget"
          onClick="onClick"
        />
        <StartStopButton
          control="stop"
          urlRoute="bodyTarget"
          onClick="onClick"
        />
      </>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
