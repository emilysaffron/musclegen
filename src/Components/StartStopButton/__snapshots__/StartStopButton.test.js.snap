// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartStopButton Snapshot 1`] = `
Array [
  <a
    href="/workout"
    onClick={[Function]}
  >
    <button
      className="css-1ezathv"
      onClick="onClick"
    >
       Start
    </button>
  </a>,
  <a
    href="/bodytarget"
    onClick={[Function]}
  >
    <button
      className="css-1ezathv"
      onClick="onClick"
    >
      Stop
    </button>
  </a>,
]
`;
